test/common/testlib.py

```diff
@@ -1095,7 +1032,131 @@ def copy_cores(self, title, label=None):
                         attach(dest)
 
 
-some_failed = False
+class MachineCase(BaseCase):
+    def setUp(self):
```

Comment:
Must call the parent `super().setUp()`. Same for all other setUp(), setUpClass(), tearDown() and so on.